### PR TITLE
support registration green path

### PR DESCRIPTION
**Pull request by amaziagur**
This is a **partial** PR.

Please give input on mainly on TicktokClient api use

### CODE REVIEW

File: src/main/java/io/ticktok/TicktokClient.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import io.ticktok.register.RegisterClockRequest;
+import io.ticktok.rest.TicktokRestClient;
+
+import java.io.IOException;
+
+public class TicktokClient{
+
+    private String token;
+
+    public TicktokClient(String token){
```
Comment (rockem) [inline src/main/java/io/ticktok/TicktokClient.java]: how about just Ticktok?

File: src/main/java/io/ticktok/TicktokClient.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        this.token = token;
+    }
+
+    public void registerClock(String schedule, String consumerId) throws IOException {
```
Comment (rockem) [inline src/main/java/io/ticktok/TicktokClient.java]: newClock?

File: src/main/java/io/ticktok/rest/TicktokRestClient.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    private HttpClient client;
+    private final String domain;
+
+    public TicktokRestClient(String domain){
+        this.domain = domain;
+        this.client = HttpClientBuilder.create().build();
+    }
+
+    private void handleBodyCall(HttpEntityEnclosingRequestBase request, String entry) throws IOException {
+        request.setEntity(new StringEntity(entry));
```
Comment (rockem) [inline src/main/java/io/ticktok/rest/TicktokRestClient.java:28]: you can use the simpler Request.post, look in AppDriver in ticktok tests

File: src/test/java/io/ticktok/TicktokClientTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void shouldRegisterNewClock() throws IOException {
+        new TicktokClient(TOKEN).registerClock("every.5.seconds", "ticktok");
+        assertThat(ticktockServiceStub.getToken(), is(TOKEN));
+        assertThat(ticktockServiceStub.getBody(), is(expectedBody()));
+    }
+
+    private String expectedBody() {
+        return new Gson().toJson(new RegisterClockRequest("ticktok", "every.5.seconds"));
```
Comment (rockem) [inline src/test/java/io/ticktok/TicktokClientTest.java]: Mind you I removed the consumer id field

**Review by rockem — CHANGES_REQUESTED** [review]

File: src/main/java/io/ticktok/TicktokClient.java (hunk 1 above)
Comment (amaziagur) [reply src/main/java/io/ticktok/TicktokClient.java]: excepted

File: src/main/java/io/ticktok/rest/TicktokRestClient.java (hunk 3 above)
Comment (amaziagur) [reply src/main/java/io/ticktok/rest/TicktokRestClient.java:28]: for now it works for me the use of `org.apache.httpcomponents.httpclient`

File: src/test/java/io/ticktok/TicktokClientTest.java (hunk 4 above)
Comment (amaziagur) [reply src/test/java/io/ticktok/TicktokClientTest.java]: removed from code, but we need to think of a better way to notify clients libs of changes in api's

File: src/main/java/io/ticktok/rest/TicktokRestClient.java (hunk 3 above)
Comment (rockem) [reply src/main/java/io/ticktok/rest/TicktokRestClient.java:28]: Request.post is a simpler API from the same library

Comment (amaziagur) [reply src/main/java/io/ticktok/rest/TicktokRestClient.java:28]: maybe, but once ill add other types of requests, i need very little code to add.

File: src/main/java/io/ticktok/Ticktok.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    public Ticktok(TicktokOptions options){
+        this.options = options;
+    }
+
+    public void newClock(String schedule) throws IOException {
+        new TicktokRestClient(this.options).post(handleBody(schedule), TicktokApi.REGISTER_NEW_CLOCK);
+    }
+
+    private String handleBody(String schedule) {
```
Comment (rockem) [inline src/main/java/io/ticktok/Ticktok.java:21]: looks like createRegisterClockRequestFor(

**Review by rockem — CHANGES_REQUESTED** [review]

**Review by rockem — APPROVED** [review]

**Pull request merged**